sound/soc/sof/core.c

```diff
@@ -246,8 +246,17 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	if (plat_data->sof_probe_complete)
 		plat_data->sof_probe_complete(sdev->dev);
 
+	/* If registering certain clients fails, unregister the previously registered clients. */
```

Comment:
comment on commit message: missing reviewed-by and Dave's signed-off by (if he sends the patches)?